internal/worker/kv_status.go

```diff
@@ -89,3 +91,94 @@ func NewStatusKVPublisher(s events.Stream, log *logrus.Logger, opts ...kv.Option
 		log: log,
 	}
 }
+
+type taskState int
+
+const (
+	notStarted    taskState = iota
```

Comment:
Could this use the conditionKind states instead? The two additional states we'd have to add are - `orphaned`, `inderminate` .. thinking out loud, at some point we'd want to share these methods with other controllers as well.

Reply:
`tl;dr` I don't think these belong in ConditionOrc, but potentially in the event handling library once we break it out of `hollow-toolbox`?

I don't think these states belong in ConditionOrc _per se_. The ConditionOrc states are things we would serialize, as in "this task is `active`." `Orphaned` wouldn't get serialized because the controller doing the work crashed or was interrupted and there's no independent observer to see that crash and update the event in the KV. `Indeterminate` is a marker for "I got an error trying to check on this." That is likely an error trying to reach NATS, but also the theoretically possible case of a process writing malformed junk to a valid key.

Reply:
ack, thanks for the explanation